src/components/meals/Meals.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import styles from '../../styles';
+import {
+  MealCategories,
+  MealInput,
+  MealList,
+  MealsComponentProps,
+  View,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  FlatList,
+} from '../../models/meals.model';
+
+const MAX_ITEMS = 3;
+
+const Meals: React.FC<MealsComponentProps> = ({ mealCategories }) => {
+  const [mealInput, setMealInput] = useState<Record<MealCategories, MealInput>>({
+    [MealCategories.BREAKFAST]: '',
+    [MealCategories.LUNCH]: '',
+    [MealCategories.DINNER]: '',
+  });
+  const [mealList, setMealList] = useState<Record<MealCategories, MealList>>({
+    [MealCategories.BREAKFAST]: [],
+    [MealCategories.LUNCH]: [],
+    [MealCategories.DINNER]: [],
+  });
+  const [enteredText, setEnteredText] = useState<Record<MealCategories, MealInput>>({
+    [MealCategories.BREAKFAST]: '',
+    [MealCategories.LUNCH]: '',
+    [MealCategories.DINNER]: '',
+  });
+
+  const onMealInputChange = (text: MealInput, category: MealCategories) => {
+    setMealInput((prev) => ({
+      ...prev,
+      [category]: text,
+    }));
+  };
+
+  const handleMealInputPress = (category: MealCategories) => {
+    const currentList = [...mealList[category], mealInput[category]];
+
+    if (currentList.length <= MAX_ITEMS) {
+      setEnteredText((prev) => ({
+        ...prev,
+        [category]: currentList,
+      }));
+
+      setMealList((prev) => ({
+        ...prev,
+        [category]: currentList,
+      }));
+    }
+
+    setMealInput((prev) => ({
+      ...prev,
+      [category]: '',
+    }));
+  };
+
+  return (
+    <View>
```

Comment:
Can we create a component for a meal? Which will consist the input + the items added
<img width="446" alt="image" src="https://github.com/BinaHovav/MyMeals-Frontend/assets/33668592/49158fc0-a00b-421d-bb17-0a03a5973231">

Reply:
no problem!